tests/testthat/test-format_data.R

```diff
@@ -0,0 +1,326 @@
+library(testthat)
+library(dplyr)
+library(PKNCA)
+
+test_that("create_conc generates correct dataset", {
+  # Generate sample ADNCA data
+  ADNCA <- data.frame(
+    STUDYID = rep(1, 10),
+    USUBJID = rep(1:2, each = 5),
+    PCSPEC = rep("Plasma", 10),
+    DRUG = rep("DrugA", 10),
+    ANALYTE = rep("Analyte1", 10),
+    AFRLT = seq(0, 9),
+    AVAL = runif(10)
+  )
+
+  # Call create_conc
+  df_conc <- create_conc(ADNCA,
+                         group_columns = c("STUDYID", "USUBJID", "PCSPEC", "DRUG", "ANALYTE"),
+                         time_column = "AFRLT")
+
+  # Test if df_conc is a data frame
+  expect_s3_class(df_conc, "data.frame")
+
+  # Test if df_conc has the correct columns
+  expect_true(all(c("STUDYID", "USUBJID", "PCSPEC", "DRUG", "ANALYTE",
+                    "AFRLT", "AVAL", "conc_groups", "TIME", "IX") %in% colnames(df_conc)))
+
+  # Test if df_conc is correctly grouped and arranged
+  expect_equal(df_conc$TIME, df_conc$AFRLT)
+  expect_equal(df_conc$IX, rep(1:5, 2))
+
+  # Test if df_conc can be used with PKNCAconc by testing its output
+  expect_s3_class(
+    PKNCAconc(
+      df_conc,
+      formula = AVAL ~ TIME | STUDYID + PCSPEC + DRUG + USUBJID / ANALYTE,
+      exclude_half.life = "exclude_half.life",
+      time.nominal = "NFRLT"
+    ),
+    "PKNCAconc"
+  )
```

Comment:
issue: In this case, if `df_conc` was somehow wrong, we would expect an error from `PKNCAconc` function, so I would wrap that in `expect_no_error` to know exactly if a) an error occured or b) `PKNCAconc` returned something strange.
```suggestion
  expect_no_error({
    expect_s3_class(
      PKNCAconc(
        df_conc,
        formula = AVAL ~ TIME | STUDYID + PCSPEC + DRUG + USUBJID / ANALYTE,
        exclude_half.life = "exclude_half.life",
        time.nominal = "NFRLT"
      ),
      "PKNCAconc"
    )
  })
```